src/main/java/com/mraof/minestuck/data/MinestuckStartingModusProvider.java

```diff
@@ -0,0 +1,65 @@
+package com.mraof.minestuck.data;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.mraof.minestuck.inventory.captchalogue.ModusType;
+import com.mraof.minestuck.inventory.captchalogue.ModusTypes;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.DataProvider;
+import net.minecraft.data.HashCache;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.mraof.minestuck.util.StartingModusManager.PATH;
+
+public class MinestuckStartingModusProvider implements DataProvider
+{
+	private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
+	private final DataGenerator generator;
+	
+	public MinestuckStartingModusProvider(DataGenerator generator)
+	{
+		this.generator = generator;
+	}
+	
+	private List<ModusType<?>> createDefaultModusTypes()
+	{
+		LinkedList<ModusType<?>> modusTypes = new LinkedList<>();
+		modusTypes.add(ModusTypes.STACK);
+		modusTypes.add(ModusTypes.QUEUE);
+		return modusTypes;
+	}
+	
+	@Override
+	public void run(HashCache pCache) throws IOException
+	{
+		Path path = this.generator.getOutputFolder().resolve(PATH);
```

Comment:
Tested it out, and discovered that the path is missing parts, causing it to generate the file in "generated/resources/minestuck/config" folder instead of  "generated/resources/data/minestuck/minestuck/config"

Reply:
I couldnt reproduce it

Reply:
Looks like you fixed it in the most recent commit. (Gonna test it tonight and see if it does what it should)
If you couldn't reproduce it before, it might be due to the cache problem described in the other review comment.